src/documentation/sdks/euphoria.md

```diff
@@ -18,72 +18,530 @@ See the License for the specific language governing permissions and
+```
+Or from `PCollection`. That is especially useful when some of many Beam's IO are utilized as data inputs.
+```java
+PCollection<T> inputPCollection = ...
+
+BeamFlow flow = BeamFlow.of(inputPCollection);
+```
+### Datasets
+Euphoria use concept of 'Datasets' to describe flows of data between operators. This concept is similar to Beam's `PCollection` and can be converted both ways through `BeamFlow`.
+```java
+PCollection<T> someCollection = ...
+
+// PCollection -> Dataset
+Dataset<T> dataset = flow.wrapped(someCollection);
+
+//And now back: Dataset -> PCollection
+PCollection<T> collection = flow.unwrapped(dataset);
+```
+One should not mix different instances of `BeamFlow` when wrapping/unwrapping `PCollections`.
+
+### Inputs and Outputs
+Input data can supplied through Beams IO into `PCollection`, the same way as in Beam, and latter wrapped by `BeamFlow` into `Dataset`.
+
+```java
+PCollection<String> input =
+  pipeline
+    .apply(Create.of("mouse", "rat", "elephant", "cat", "X", "duck"))
+    .setTypeDescriptor(TypeDescriptor.of(String.class));
+
+Dataset<String> dataset = flow.wrapped(input);
+```
+Output can be treated the same way as inputs, last `Dataset` is converted to `Pcollection` and dumped to appropriate IO.
+
+### Adding Operators
+Real power of Euphoria API is in its [operators suite](#operators-reference). Once we get our hand on `Dataset` we are able to create and connect operators. Each Operator consumes one or more input and produces one output
+`Dataset`. Lets look at simple `MapElements` example.
+
+```java
+Dataset<Integer> input = ...
+
+Dataset<String> mappedElements =
+  MapElements
+    .named("Int2Str")
+    .of(input)
+    .using(String::valueOf)
+    .output();
+```
+The operator consumes `input`, it applies given lambda expression (`String::valueOf`) on each element of `input` and returns mapped `Datatset`. Developer is guided through series of steps when creating operator so the declaration of an operator is straightforward. To start building operator just wrote its name and '.'. Your IDE will give you hits.
+
+First step to build any operator is to give it a name through `named()` method. This is optional but recommended step. The name is propagated through system and can latter be used when debugging.
+
+### Coders and Types
+Beam's Java SDK requires developers to supply `Coder` for custom element type in order to have a way of materializing elements. Euphoria integrates [Kryo](https://github.com/EsotericSoftware/kryo) as default way of serialization. Recommended way of using [Kryo](https://github.com/EsotericSoftware/kryo) is to register all the types which will Kryo serialize. Sometimes it is also a good idea to register Kryo serialisers of its own too. Euphoria allows you to do that by implementing your own `KryoRegistrar` and using it as input to `RegisterCoders`. The `RegisterCoders` can also be used as convenient way of supplying Beam `Coder` for specified element types.
```

Comment:
*serializers

`and using it as an input for`

Reply:
Fixed.